File: tests/fixtures/typescript/processor.ts
// DataProcessor.ts — core data processing with inheritance and cross-file imports

export interface ProcessorConfig {
  requiredFields?: string[];
  strict?: boolean;
}

export interface Transformer {
  transform(data: Record<string, unknown>): Record<string, unknown>;
}

export class DataProcessor {
  private transformers: Transformer[] = [];

  constructor(protected config: ProcessorConfig = {}) {}

  addTransformer(t: Transformer): void {
    this.transformers.push(t);
  }

  process(data: Record<string, unknown>): Record<string, unknown> {
    let result = { ...data };
    for (const t of this.transformers) {
      result = t.transform(result);
    }
    return result;
  }

  validate(data: Record<string, unknown>): boolean {
    const required = this.config.requiredFields ?? [];
    return required.every((f) => f in data);
  }
}

export class JSONProcessor extends DataProcessor {
  process(data: Record<string, unknown>): Record<string, unknown> {
    const parsed = JSON.parse(JSON.stringify(data));
    return super.process(parsed);
  }
}

export function createDefaultProcessor(): DataProcessor {
  // cross-file import — tests call resolution
  return new DataProcessor({ requiredFields: ["id", "name"] });
}

File: tests/fixtures/java/Processor.java
// Processor.java — core data processing with inheritance and interface implementation

package fixtures;

import java.util.*;

public class DataProcessor {
    protected ProcessorConfig config;
    protected List<Transformer> transformers;

    public DataProcessor(ProcessorConfig config) {
        this.config = config;
        this.transformers = new ArrayList<>();
    }

    public void addTransformer(Transformer t) {
        this.transformers.add(t);
    }

    public Map<String, Object> process(Map<String, Object> data) {
        Map<String, Object> result = new HashMap<>(data);
        for (Transformer t : this.transformers) {
            result = t.transform(result);
        }
        return result;
    }

    public boolean validate(Map<String, Object> data) {
        for (String field : config.getRequiredFields()) {
            if (!data.containsKey(field)) {
                return false;
            }
        }
        return true;
    }
}

class JSONProcessor extends DataProcessor {
    public JSONProcessor(ProcessorConfig config) {
        super(config);
    }

    @Override
    public Map<String, Object> process(Map<String, Object> data) {
        return super.process(new HashMap<>(data));
    }
}
